Data Prepper documentation updates: autogeneration campaign (#7707)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/parse_xml.md

-->

| Option | Required | Type | Description |
| :--- | :--- | :--- | :--- | 
| `source` | No | String | Specifies which `event` field to parse. |
| `destination` | No | String | The destination field of the parsed XML. Defaults to the root of the `event`. Cannot be `""`, `/`, or any white-space-only string because these are not valid `event` fields. |
| `pointer` | No | String | A JSON pointer to the field to be parsed. The value is null by default, meaning that the entire `source` is parsed. The `pointer` can access JSON array indexes as well. If the JSON pointer is invalid, then the entire `source` data is parsed into the outgoing `event` object. If the key that is pointed to already exists in the `event` object and the `destination` is the root, then the pointer uses the entire path of the key. |
| `parse_when` | No | String | Specifies under what conditions the processor should perform parsing. Default is no condition. Accepts a Data Prepper expression string following the [Data Prepper Expression Syntax]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/). |

> | `overwrite_if_destination_exists` | No | Boolean | If set to `true` then overwrite the destination. Set to `false` to prevent changing a destination value that exists. Defaults to `true`. |

Reviewer: ```suggestion
| `overwrite_if_destination_exists` | No | Boolean | If set to `true` then overwrite the destination. Set to `false` to prevent changing a destination value that exists. Defaults to `true`. |
```

> | `tags_on_failure` | No | String | A list of strings that specify the tags to be set if the processor fails or an unknown exception occurs while parsing.

Reviewer: ```suggestion
| `tags_on_failure` | No | String | A list of strings specifying the tags to be set in the event that the processor fails or an unknown exception occurs during parsing.
```
